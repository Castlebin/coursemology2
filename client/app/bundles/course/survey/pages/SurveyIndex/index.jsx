import React, { PropTypes } from 'react';
import { connect } from 'react-redux';
import { defineMessages, FormattedMessage } from 'react-intl';
import Subheader from 'material-ui/Subheader';
import TitleBar from 'lib/components/TitleBar';
import { fetchSurveys } from 'course/survey/actions/surveys';
import surveyTranslations from 'course/survey/translations';
import { surveyShape } from 'course/survey/propTypes';
import SurveysTable from './SurveysTable';
import NewSurveyButton from './NewSurveyButton';

const translations = defineMessages({
  noSurveys: {
    id: 'course.surveys.SurveyIndex.noSurveys',
    defaultMessage: 'No surveys have been created.',
  },
});

class SurveyIndex extends React.Component {
  static propTypes = {
    dispatch: PropTypes.func.isRequired,
    surveys: PropTypes.arrayOf(surveyShape),
    params: PropTypes.shape({
      courseId: PropTypes.string.isRequired,
    }),
  };

  componentDidMount() {
    const { dispatch } = this.props;
    dispatch(fetchSurveys());
  }

  render() {
    const { surveys, params: { courseId } } = this.props;
    return (
      <div>
        <TitleBar
          title={<FormattedMessage {...surveyTranslations.surveys} />}
        />
        {
          surveys.length > 0 ?
            <SurveysTable {...{ courseId }} /> :
            <Subheader><FormattedMessage {...translations.noSurveys} /></Subheader>
        }
        <NewSurveyButton />
      </div>
    );
  }
}

export default connect(state => state)(SurveyIndex);
