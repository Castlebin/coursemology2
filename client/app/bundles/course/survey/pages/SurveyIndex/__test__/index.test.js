import React from 'react';
import { shallow } from 'enzyme';
import storeCreator from 'course/survey/store';
import SurveyIndex from '../index';

describe('<SurveyIndex />', () => {
  it('renders the index page', async () => {
    const store = storeCreator({ surveys: {} });

    const indexPage = shallow(
      <SurveyIndex params={{ courseId }} />,
      {
        context: { intl, store, muiTheme },
        childContextTypes: { muiTheme: React.PropTypes.object, intl: intlShape },
      }
    );

    expect(indexPage).toMatchSnapshot();
  });
});
