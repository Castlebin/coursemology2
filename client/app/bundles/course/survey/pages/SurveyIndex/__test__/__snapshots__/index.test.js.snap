// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SurveyIndex /> renders the index page 1`] = `
<SurveyIndex
  canCreate={false}
  deleteConfirmation={
    Object {
      "onConfirm": [Function],
      "open": false,
    }
  }
  dispatch={[Function]}
  form={Object {}}
  notification={Object {}}
  params={
    Object {
      "courseId": "1",
    }
  }
  questionForm={
    Object {
      "disabled": false,
      "formTitle": "",
      "initialValues": Object {},
      "onSubmit": [Function],
      "visible": false,
    }
  }
  responseForm={
    Object {
      "isLoading": false,
      "response": Object {},
      "survey": Object {},
    }
  }
  results={
    Object {
      "sections": Array [],
      "survey": Object {},
    }
  }
  surveyForm={
    Object {
      "disabled": false,
      "formTitle": "",
      "initialValues": Object {},
      "onSubmit": [Function],
      "visible": false,
    }
  }
  surveys={Array []}
/>
`;
